blockdag/validate.go

```diff
@@ -838,6 +854,11 @@ func (dag *BlockDAG) checkConnectToPastUTXO(block *blockNode, pastUTXO UTXOSet,
 			return nil, err
 		}
 
+		err = checkDoubleSpendsWithBlockPast(pastUTXO, transactions)
```

Comment:
Is there any chance checkDoubleSpendsWithBlockPast makes ensureNoDuplicateTx redundant?

Reply:
ensureNoDuplicateTx checks another thing (that you do not have a transaction that has the same ID as another tx in the UTXO, maybe because of hash collision), but maybe you talked about the duplicate transaction check in checkBlockSanity. But if we'll remove it we'll not get errors for hash collisions in the same block.

Reply:
Hmmmm... actually, it seems like this is implicitly validated at least two more times: In CheckBlockMass and when validating the signatures, since both require all relevant UTXOs to exist in the pastUTXO.
In that case, do we really want to validate this explicitly?